fop/fom.c

```diff
@@ -335,11 +337,28 @@ M0_INTERNAL bool m0_fom_invariant(const struct m0_fom *fom)
 		_0C(fom->fo_service->rs_type == fom->fo_type->ft_rstype);
 }
 
+
+static void m0_fom_in_deadlock(const struct m0_fom *fom0)
+{
+	if (!M0_IN(fom0->fo_type->ft_id, (M0_CAS_GET_FOP_OPCODE,
```

Comment:
M0_CAS_GET_FOP_OPCODE, M0_CAS_CUR_FOP_OPCODE and M0_CAS_REP_FOP_OPCODE are unnecessary here as they don't use tx. M0_CAS_GCW_FOP_OPCODE and M0_CAS_GCF_FOP_OPCODE also can be removed. PUT and DEL are enough here.